msticpy/data/uploaders/splunk_uploader.py

```diff
@@ -116,7 +123,7 @@ def _post_data(
     def upload_df(  # type: ignore
         self,
         data: pd.DataFrame,
-        table_name: Optional[str],
+        sourcetype: Optional[str],
```

Comment:
given that you are changing the parameter name, I would try to keep backward compatibility and maybe opt for json as the default. I have a mild preference for snake-case compound words like this - source_type vs sourcetype. I'm not super concerned about that though
```python

    source_type = kwargs.get("source_type", kwargs.get("table_name", "json"))
```